File: system/monolith/dotnet/Core/Dtos/PlaceOrderRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MyCompany.MyShop.Monolith.Core.Dtos;

public class PlaceOrderRequest
{
    [Required(ErrorMessage = "SKU must not be empty")]
    public string? Sku { get; set; }

    [Required(ErrorMessage = "Quantity must not be empty")]
    [Range(1, int.MaxValue, ErrorMessage = "Quantity must be positive")]
    public int? Quantity { get; set; }

    [Required(ErrorMessage = "Country must not be empty")]
    public string? Country { get; set; }

    public string? CouponCode { get; set; }
}

File: system/monolith/java/src/main/java/com/mycompany/myshop/monolith/core/dtos/PlaceOrderRequest.java
package com.mycompany.myshop.monolith.core.dtos;

import com.mycompany.myshop.monolith.core.validation.TypeValidationMessage;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Data;

@Data
public class PlaceOrderRequest {
    @NotBlank(message = "SKU must not be empty")
    private String sku;

    @NotNull(message = "Quantity must not be empty")
    @Positive(message = "Quantity must be positive")
    @TypeValidationMessage("Quantity must be an integer")
    private Integer quantity;

    @NotBlank(message = "Country must not be empty")
    private String country;

    private String couponCode;
}
